app/src/org/commcare/views/dialogs/CommCareAlertDialog.java

```diff
@@ -19,16 +19,20 @@
     protected DialogInterface.OnCancelListener cancelListener;
     private DialogInterface.OnDismissListener dismissListener;
     protected View view;
-    // false by default, can be overridden by calling setOnCancelListener(), or by subclass
-    // definitions if desired
+    // false by default, can be overridden if desired
     protected boolean isCancelable;
 
     public void finalizeView() {
         dialog.setCancelable(isCancelable);
-        if (isCancelable) {
+        if (cancelListener != null) {
```

Comment:
Yea, the use-case for that was actually added in this PR. Line 1284 of `CommCareHomeActivity` above calls `makeCancelable()`, but doesn't set an `onCancelListener()`. I did it this way because when a dialog is canceled by the user, BOTH `onCancel()` and `onDismiss()`actually get called (the reverse is not true). So I wanted to only add an `onDismissListener` to the About CommCare dialog, to prevent the logic from getting called twice in the case of the dialog being canceled. But I still want the dialog to be cancelable, hence allowing a dialog to be cancelable without a listener.